ts/input/tex/ParseUtil.ts

```diff
@@ -450,6 +485,26 @@ namespace ParseUtil {
     return s1 + s2;
   }
 
+  /**
+   * Report an error if there are too many macro substitutions.
+   * @param {TexParser} parser The current TeX parser.
+   * @param {boolean} isMacro  True if we are substituting a macro, false for environment.
+   */
+  export function checkMaxMacros(parser: TexParser, isMacro: boolean = true) {
+    if (++parser.macroCount <= parser.configuration.options['maxMacros']) {
+      return;
+    }
+    if (isMacro) {
+      throw new TexError('MaxMacroSub1',
+                         'MathJax maximum macro substitution count exceeded; ' +
+                         'is here a recursive macro call?');
+    } else {
```

Comment:
No need for the else.

Reply:
I know that you prefer to reduce indentation when possible, but the indentation sometimes has meaning.  In this case, the function will produce one of two error messages, neither of which is more important than the other.  The indentation of both shows that they are comparable alternatives.  Leaving out the "else" would make the second be the "main" message and the first be a subordinate one.  It makes the `isMacro` variable a break0out condition rather than a toggle that determines which of two equal things is being selected.  This structure preserves the logic of the function better than having the two messages at different indentations, which gives the second a prominence it doesn't deserve.

Reply:
Unless you strongly object, I'd like to keep it as is.